torch/distributed/distributed_c10d.py

```diff
@@ -487,7 +502,13 @@ def _new_process_group_helper(world_size,
             _pg_map[pg] = (Backend.NCCL, store)
             _pg_names[pg] = group_name
         else:
-            raise RuntimeError("Unsupported distributed backend by group")
+            pg = getattr(Backend, backend.upper())(
+                prefix_store,
+                rank,
+                world_size,
+                group_name)
```

Comment:
Does this mean all 3rd party backend must use the same signature? What about `timeout`? and why do we need to pass in group_name here?

Reply:
@mrshenli understand your concern. At my local first version, I added timeout and no group_name, but later removed it as timeout is not necessary for our c10d backend. 

I would add timeout back for all 3rd party backend.